modules/runner/src/main/java/org/apache/ignite/runner/internal/storage/DistributedConfigurationStorage.java

```diff
@@ -18,38 +18,86 @@
 package org.apache.ignite.runner.internal.storage;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
 import org.apache.ignite.configuration.storage.ConfigurationStorage;
 import org.apache.ignite.configuration.storage.ConfigurationStorageListener;
 import org.apache.ignite.configuration.storage.Data;
 import org.apache.ignite.configuration.storage.StorageException;
+import org.apache.ignite.lang.util.SerializationUtils;
+import org.apache.ignite.metastorage.common.Cursor;
+import org.apache.ignite.metastorage.common.Entry;
+import org.apache.ignite.metastorage.common.Key;
 import org.apache.ignite.metastorage.internal.MetaStorageManager;
 
 public class DistributedConfigurationStorage implements ConfigurationStorage {
+    // TODO Distributed prefix. Will be replaced when ENUM with configuration type will be presented.
+    // https://issues.apache.org/jira/browse/IGNITE-14476
+    private static String DISTRIBUTED_PREFIX = "distributed";
+
     private final MetaStorageManager metaStorageMgr;
 
     public DistributedConfigurationStorage(MetaStorageManager metaStorageMgr) {
         this.metaStorageMgr = metaStorageMgr;
     }
 
-    @Override public Data readAll() throws StorageException {
-        return null;
+    /** Change listeners. */
+    private List<ConfigurationStorageListener> listeners = new ArrayList<>();
+
+    /** Storage version. */
+    private AtomicLong version = new AtomicLong(0);
+
+    /** {@inheritDoc} */
+    @Override public synchronized Data readAll() throws StorageException {
+
+        Cursor<Entry> cur = metaStorageMgr.service().range(new Key(DISTRIBUTED_PREFIX + "."), new Key(DISTRIBUTED_PREFIX + (char)('.' + 1)));
+
+        HashMap<String, Serializable> data = new HashMap<>();
+
+        for (Entry entry : cur)
+            data.put(entry.key().toString().replace(DISTRIBUTED_PREFIX + ".", ""), (Serializable)SerializationUtils.fromBytes(entry.value()));
+
+        // storage revision 0?
+        return new Data(data, version.get(), 0);
```

Comment:
I'm not sure whether storageRevision is metastorage revision or key updateCounter. In any case it won't be 0 here. it'll be max revision or updateCounter that we saw within all entries that match r`range(new Key(DISTRIBUTED_PREFIX + "."), new Key(DISTRIBUTED_PREFIX + (char)('.' + 1)));`